realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -407,7 +411,7 @@ public boolean hasPrimaryKey() {
      */
     public Set<String> getFieldNames() {
         int columnCount = (int) table.getColumnCount();
-        Set<String> columnNames = new TreeSet<String>();
+        Set<String> columnNames = new LinkedHashSet<>(columnCount);
```

Comment:
Why the switch to `LinkedHashSet`? What operations are we trying to optimize for?